test/inductor/test_torchinductor.py

```diff
@@ -243,6 +244,52 @@ def run(*ex, **kwargs):
                 if exact_dtype:
                     assert correct_val.dtype == actual_val.dtype
 
+    if check_gradient:
+
+        def gather_leaf_tensors(args, kwargs):
+            leaf_tensors = []
+            args, args_spec = tree_flatten(args)
+            kwargs, kwargs_spec = tree_flatten(kwargs)
+            args = args + kwargs
+            for arg in args:
+                if not isinstance(arg, torch.Tensor):
+                    continue
+                if arg.requires_grad:
+                    leaf_tensors.append(arg)
+            return leaf_tensors
+
+        def compute_grads(args, kwrags, results):
```

Comment:
Nit: maybe pull to top level? It took me a second to realize that it wasn't calling itself in the compute_grads below.

Reply:
ok.